tests_e2e/tests/agent_update/self_update.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+
+# Microsoft Azure Linux Agent
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+from pathlib import Path
+from threading import RLock
+
+from assertpy import fail
+
+import azurelinuxagent
+from tests_e2e.tests.lib.agent_test import AgentVmTest
+from tests_e2e.tests.lib.agent_test_context import AgentVmTestContext
+from tests_e2e.tests.lib.logging import log
+from tests_e2e.tests.lib.retry import retry_if_false
+from tests_e2e.tests.lib.shell import run_command
+
+
+class SelfUpdateBvt(AgentVmTest):
+    """
+    This test case is to verify that the agent can update itself to the latest version using self-update path when vm not enrolled to RSM updates
+    """
+
+    def __init__(self, context: AgentVmTestContext):
+        super().__init__(context)
+        self._ssh_client = self._context.create_ssh_client()
+        self._test_version = "2.8.9.9"
```

Comment:
No specific reason, it could be some random version, but I took 2.8 because some distros preinstall 2.7. So, it should be greater than that. 

I have a setup in rsm update where we modify the daemon version to some min value, so that deamon version won't get in the way. Since I disable the rsm update scenario today, I had to go with 2.8